tests/bluetooth/bsim_bt/bsim_test_gatt_caching/src/common.c

```diff
@@ -39,16 +40,25 @@ void backchannel_init(void)
 
 void backchannel_sync_send(void)
 {
-	/* Dummy message */
-	uint8_t sync_msg[] = { 'A' };
+	uint8_t sync_msg[MSG_SIZE] = { get_device_nbr() };
 
 	printk("Sending sync\n");
 	bs_bc_send_msg(CHANNEL_ID, sync_msg, ARRAY_SIZE(sync_msg));
 }
 
 void backchannel_sync_wait(void)
 {
-	while (!bs_bc_is_msg_received(CHANNEL_ID)) {
+	uint8_t sync_msg[MSG_SIZE];
+
+	while (true) {
+		if (bs_bc_is_msg_received(CHANNEL_ID) > 0) {
+			bs_bc_receive_msg(CHANNEL_ID, sync_msg, ARRAY_SIZE(sync_msg));
+			if (sync_msg[0] != get_device_nbr()) {
```

Comment:
Just out of curiosity since I can't find it documented anywhere: what is `nbr` in this context? :D

Reply:
`get_device_nbr()` returns the "Overall device number in the simulation", which is set by the "-d" or "-gd" option when running the test executable. In typical C fashion, all vowels are removed.

Reply:
Had to search through to `babblesim/components/libUtilv1/src/bs_cmd_line_typical.h` to find the definition.

Reply:
Ah, yes
```
    { false, true  , false, "d",       "device_number",        'u', (void*)&args->device_nbr,NULL,         "Device number (for this phy)"}
```

I think this is the first time I've encountered the `number`->`nbr` abbreviation :D